lib/Sema/CSSimplify.cpp

```diff
@@ -6026,7 +6159,11 @@ ConstraintSystem::simplifyCheckedCastConstraint(
       addConstraint(ConstraintKind::Subtype, toType, fromType,
                     getConstraintLocator(locator));
     }
-      
+
+    if (auto *fix = fixForCheckedConvertibleTypes()) {
+      (void)recordFix(fix);
```

Comment:
Could you please leave a note here why result type is being ignored? It's good to have for posterity since future us/others might not read `fixForCheckedConvertibleTypes` to understand what is going on here.

Reply:
Comments added :))